### PR TITLE
feat(atom/button): add rounded button support

### PR DESCRIPTION
**Pull request by andresz1**
## atom/button
**TASK**: None

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description, Motivation and Context
This PR adds `isRounded` prop to `AtomButton`, if enabled rounded borders variation will be use.

### Screenshots - Animations
Screenshot 2020-10-09 at 14 31 50

### CODE REVIEW

**Comment by turolopezsanabria** [comment]
Do you think we could avoid Boolean for this prop? Maybe imagine this one to be used as "shape" or something similar, as if it has multiple options? square, rounded, circular... 

I really like this article @andresin87 shared time ago
https://spicefactory.co/blog/2019/03/26/how-to-avoid-the-boolean-trap-when-designing-react-components/

File: components/atom/button/src/index.scss [hunk 1]
```diff
@@ -345,6 +349,10 @@ $icon-stroke-color: currentColor !default;
     }
   }
 
+  &--rounded {
+    border-radius: $bdrs-atom-button-rounded;
```
Comment (jcger) [inline components/atom/button/src/index.scss]: shouldn't this be using the rounded border token and thats it?

Comment (andresin87) [reply components/atom/button/src/index.scss]: Maybe is 4 combining both usages integrated

Comment (andresz1) [reply components/atom/button/src/index.scss]: I don't think so cuz the width and height of the button are different (see [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/border-radius))

**Comment by andresz1** [comment]
> Do you think we could avoid Boolean for this prop? Maybe imagine this one to be used as "shape" or something similar, as if it has multiple options? square, rounded, circular...
> 
> I really like this article @andresin87 shared time ago
> https://spicefactory.co/blog/2019/03/26/how-to-avoid-the-boolean-trap-when-designing-react-components/

@turolopezsanabria I thought that maybe for this case was ok due I couldn't think of any other variation related. So what about `shape` prop with possible values to `square` (default) and `rounded` (this case). If no `children` is provided will be displayed as a `IconButton` as it is ATM.

**Comment by andresin87** [comment]
> > Do you think we could avoid Boolean for this prop? Maybe imagine this one to be used as "shape" or something similar, as if it has multiple options? square, rounded, circular...
> > I really like this article @andresin87 shared time ago
> > [spicefactory.co/blog/2019/03/26/how-to-avoid-the-boolean-trap-when-designing-react-components](https://spicefactory.co/blog/2019/03/26/how-to-avoid-the-boolean-trap-when-designing-react-components/)
> 
> @turolopezsanabria I thought that maybe for this case was ok due I couldn't think of any other variation related. So what about `shape` prop with possible values to `square` (default) and `rounded` (this case). If no `children` is provided will be displayed as a `IconButton` as it is ATM.

prefer not to use boolean props 4 that case

File: components/atom/button/src/index.scss (hunk 1 above)
Comment (turolopezsanabria) [reply components/atom/button/src/index.scss]: right, as @andresz1 comments, `rounded` token is actually an oval if the w and h are different

**Comment by turolopezsanabria** [comment]
sure! `shape` for the name of the prop is perfect.

I suggest these values:

- squared (a central button of a group for instance)
- rounded (not circular, but actually rounded, as 8px in Fotocasa for instance) this is the default value is the prop is unchanged
- circular (fully rounded)

**Comment by andresz1** [comment]
Changes made @turolopezsanabria @andresin87

**Review by stivaliserna — APPROVED** [review]

**Review by javiauso — APPROVED** [review]

**Review by andresin87 — APPROVED** [review]

**Pull request merged**